
loading train data:    : 0it [00:00, ?it/s]

loading train data:    : 3361it [00:07, 436.68it/s]

loading test data:    : 856it [00:00, 2309.96it/s]
epoch: 0
Training:    :   0%|          | 0/160 [00:00<?, ?it/s]..\evaluation\eval.py:31: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  a = (((batch[student][:, 0:C.NUM_OF_QUESTIONS] - batch[student][:, C.NUM_OF_QUESTIONS:]).sum(1) + 1)//2)[1:]
> d:\ethz\internship\adaptive-e-learning-for-educational-recommendation-system\deepknowledgetracing-dkt-pytorch\dkt\knowledgetracing\evaluation\eval.py(48)train_epoch()
-> optimizer.zero_grad()
(Pdb)
Training:    :   0%|          | 0/160 [18:29<?, ?it/s]
Traceback (most recent call last):
  File "D:\ETHz\Internship\adaptive-e-learning-for-educational-recommendation-system\DeepKnowledgeTracing-DKT-Pytorch\DKT\KnowledgeTracing\evaluation\run.py", line 41, in <module>
    model, optimizer, train_loss = eval.train(trainLoaders, model, optimizer_adgd, loss_func,device)
  File "..\evaluation\eval.py", line 90, in train
    model, optimizer, loss = train_epoch(model, trainLoaders[i], optimizer, lossFunc,device)
  File "..\evaluation\eval.py", line 48, in train_epoch
    optimizer.zero_grad()
  File "..\evaluation\eval.py", line 48, in train_epoch
    optimizer.zero_grad()
  File "C:\Users\tyu06\anaconda3\lib\bdb.py", line 88, in trace_dispatch
    return self.dispatch_line(frame)
  File "C:\Users\tyu06\anaconda3\lib\bdb.py", line 112, in dispatch_line
    self.user_line(frame)
  File "C:\Users\tyu06\anaconda3\lib\pdb.py", line 262, in user_line
    self.interaction(frame, None)
  File "C:\Users\tyu06\anaconda3\lib\pdb.py", line 357, in interaction
    self._cmdloop()
  File "C:\Users\tyu06\anaconda3\lib\pdb.py", line 322, in _cmdloop
    self.cmdloop()
  File "C:\Users\tyu06\anaconda3\lib\cmd.py", line 126, in cmdloop
    line = input(self.prompt)
  File "C:\Users\tyu06\anaconda3\lib\codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte